fabric-biome-api-v1/src/main/java/net/fabricmc/fabric/mixin/biome/StructureTemplateMixin.java

```diff
@@ -0,0 +1,81 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.biome;
+
+import java.util.Iterator;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import net.minecraft.nbt.NbtCompound;
+import net.minecraft.structure.StructurePlacementData;
+import net.minecraft.structure.StructureTemplate;
+import net.minecraft.structure.processor.StructureProcessor;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.util.math.random.Random;
+import net.minecraft.world.ServerWorldAccess;
+
+import net.fabricmc.fabric.impl.biome.StructurePlacementContext;
+
+@Mixin(StructureTemplate.class)
+public abstract class StructureTemplateMixin {
+	@Unique
+	private final ThreadLocal<StructurePlacementContext> contextThreadLocal = new ThreadLocal<>();
+
+	@Inject(method = "place", at = @At(value = "INVOKE", target = "Lnet/minecraft/structure/StructureTemplate;spawnEntities(Lnet/minecraft/world/ServerWorldAccess;Lnet/minecraft/util/math/BlockPos;Lnet/minecraft/util/BlockMirror;Lnet/minecraft/util/BlockRotation;Lnet/minecraft/util/math/BlockPos;Lnet/minecraft/util/math/BlockBox;Z)V"))
+	private void captureContext(ServerWorldAccess world, BlockPos pos, BlockPos pivot, StructurePlacementData placementData, Random random, int flags, CallbackInfoReturnable<Boolean> cir) {
+		contextThreadLocal.set(new StructurePlacementContext(
+				world,
+				placementData,
+				pos,
+				pivot
+		));
+	}
+
+	@Inject(method = "place", at = @At(value = "INVOKE", shift = At.Shift.AFTER, target = "Lnet/minecraft/structure/StructureTemplate;spawnEntities(Lnet/minecraft/world/ServerWorldAccess;Lnet/minecraft/util/math/BlockPos;Lnet/minecraft/util/BlockMirror;Lnet/minecraft/util/BlockRotation;Lnet/minecraft/util/math/BlockPos;Lnet/minecraft/util/math/BlockBox;Z)V"))
+	private void clearContext(ServerWorldAccess world, BlockPos pos, BlockPos pivot, StructurePlacementData placementData, Random random, int flags, CallbackInfoReturnable<Boolean> cir) {
+		contextThreadLocal.set(null);
+	}
+
+	@Redirect(method = "spawnEntities", at = @At(value = "INVOKE", target = "Ljava/util/Iterator;next()Ljava/lang/Object;"))
+	private Object applyEntityProcessor(Iterator<StructureTemplate.StructureEntityInfo> instance) {
+		final StructurePlacementContext context = contextThreadLocal.get();
+		StructureTemplate.StructureEntityInfo entityInfo = instance.next();
+
+		for (final StructureProcessor processor : context.placementData().getProcessors()) {
+			entityInfo = processor.process(
```

Comment:
https://github.com/Fabricators-of-Create/Porting-Lib/blob/f86f1ac1f5d7407dafb6eaadc7934da566cf29a2/extensions/src/main/java/io/github/fabricators_of_create/porting_lib/extensions/extensions/StructureProcessorExtensions.java#L9 (mojmap) This doesn't pass the StructureTemplate like StructureProcessorExtensions does meaning I can't even use this

Reply:
Hi, thanks for taking a look. What is the use case for this? The vanilla StructureProcessor does not pass it, hence why I didnt. Im quite happy to add it, just a little curious about the use case.

Reply:
Maybe it's useful to be able to access the StructureTemplate's `entityInfoList`? E.g. to constrain the number of a specific entity that spawns within a structure. Or perhaps the its `size` field could be useful for conditionally processing entities.

Reply:
@AlphaMode so, why is this useful?

Reply:
@AlphaMode up?

Reply:
Hello there, is there anything that can be done? This is such a good feature to have for mod compatibility.

Reply:
@Technici4n Btw AlphaMode said on Discord they are actually not sure about use cases for the StructureTemplate. And considering no one else has chimed in about this, I think it's safe to say we don't need to include the StructureTemplate as a parameter. That, or we could just include it anyway in case someone needs it. Regardless, it would be nice to see some activity on this PR, as the feature would be really useful!

Reply:
Ill get this PR updated, worse case we can always add an overload passing it later.